core/src/view/view.cpp

```diff
@@ -125,7 +126,7 @@ float View::getFocalLength() const {
 
 void View::setMinZoom(float minZoom) {
 
-    m_minZoom = std::max(minZoom, 0.f);
+    m_minZoom = (m_constrainToWorldBounds) ? std::max(minZoom, m_constrainZoom) : std::max(minZoom, 0.f);
```

Comment:
Something seems inconsistent here - if you set a minZoom with constrainToWorldBounds=true and then set constrainToWorldBounds=false, the minZoom is still the constrained zoom.

Reply:
I think it will be more straightforward to conditionally apply `m_minZoom` or `m_constrainZoom` directly in `setZoom`.